### ISSUE TITLE
Bidirectional wrapper with masking gives incorrect hidden states

### ISSUE THREAD

**Issue by ppasupat** [fchollet/keras#4192]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

[Code to reproduce the bug](https://gist.github.com/ppasupat/933dfcd988dcca995451d4d68a909650)

The code creates a Bidirectional SimpleRNN layer upon an Embedding layer with masking. (Using LSTM also gives the same bug.)

The hidden states of the SimpleRNN backward pass are not the same when the paddings (input 0) are placed in different positions. The forward pass seems to be fine.
